### PR TITLE
Support list by roles

### PR DESCRIPTION
**Pull request by plorenz**
* Adds more ways to generate set cursors, including from SetReadIndex values, slices, filtered cursors and union cursors. These support querying entities by role attributes.
* Removed err from a few different code paths. They were mostly triggered when encountering unknown symbols, and queries now check symbols during compliation, so those errors should never happen. If we ever want to add errors back, it would be better to do it via the symbols object that gets passed around, so it doesn't clutter the code path and we can check it on query completion.

### CODE REVIEW

File: storage/boltz/query_bolt_cursors.go [hunk 1]
```diff
@@ -72,9 +71,8 @@ type ReverseBoltCursor struct {
 	BaseBoltCursor
 }
 
-func (f *ReverseBoltCursor) Next() error {
+func (f *ReverseBoltCursor) Next() {
 	f.key, _ = f.cursor.Prev()
```
Comment (ekoby) [inline storage/boltz/query_bolt_cursors.go]: is this correct? Next() is actually Prev()?

File: storage/boltz/query_bolt_cursors.go [hunk 2]
```diff
@@ -93,10 +91,9 @@ type TypedForwardBoltCursor struct {
 	BaseBoltCursor
 }
 
-func (f *TypedForwardBoltCursor) Next() error {
+func (f *TypedForwardBoltCursor) Next() {
 	key, _ := f.cursor.Next()
```
Comment (ekoby) [inline storage/boltz/query_bolt_cursors.go]: is error ignored?

File: storage/boltz/query_bolt_cursors.go (hunk 1 above)
Comment (ekoby) [reply storage/boltz/query_bolt_cursors.go]: oops, ReverseCursor

File: storage/boltz/query_bolt_cursors.go (hunk 2 above)
Comment (plorenz) [reply storage/boltz/query_bolt_cursors.go]: Next returns key, value

File: storage/boltz/query_bolt_cursors.go (hunk 1 above)
Comment (plorenz) [reply storage/boltz/query_bolt_cursors.go]: :+1:

**Pull request merged**